AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/StateMachine/hierarchical-state-machine-swift/StateMachine.swift

```diff
@@ -39,7 +32,7 @@ actor StateMachine<
     private let resolver: AnyResolver<StateType>
 
     private(set) var currentState: StateType
-    private var subscribers: [WeakWrapper<StateAsyncSequence<StateType>>]
+    private var subscribers: AtomicDictionary<UUID, ContinuationWrapper<StateType>>
```

Comment:
This appears to be a scenario where the CurrentValueSubject could be utilized. Have we explored the possibility of using the CurrentValueSubject as a potential solution?